src/vm/linux/types.rs

```diff
@@ -1 +1,93 @@
 // SPDX-License-Identifier: Apache-2.0
+
+pub const NR_CPUID_CONFIGS: usize = 12;
+
+/// CPUID_CONFIG is designed to enumerate how the host VMM may configure the
+/// virtualization done by the Intel TDX module for a single CPUID leaf and
+/// sub-leaf. This is equivalent to `struct kvm_tdx_cpuid_config` in the kernel.
+#[derive(Debug, Default, Clone, Copy)]
+#[repr(C)]
+pub struct CpuidConfig {
+    /// EAX input value to CPUID
+    pub leaf: u32,
+    /// ECX input value to CPUID. A value of -1 indicates a CPUID leaf with
+    /// no sub-leaves.
+    pub sub_leaf: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EAX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub eax: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EBX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ebx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in ECX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ecx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EDX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub edx: u32,
+}
+
+/// Provides information about the Intel TDX module. This is equivalent to
+/// `struct kvm_tdx_capabilities` in the kernel.
+#[derive(Debug)]
+#[repr(C)]
+pub struct Capabilities {
+    /// Bitmap where if any certain bit is 0, it must be 0 in any TD's
+    /// ATTRIBUTES, which specifies various guest TD attributes. The value of
+    /// this field reflects the Intel TDX module capabilities and configuration
+    /// and CPU capabilities.
+    pub attrs_fixed0: u64,
```

Comment:
Where is this bitmap? What does each bit mean? Is this documented somewhere?

Reply:
Similar to XFAM, you can find the documentation for ATTRIBUTES in [the ABI spec](https://www.intel.com/content/www/us/en/content-details/733579/intel-tdx-module-v1-5-abi-specification.html)  on page 43. ATTRIBUTES is a 64 bit field that specifies guest attributes. The bits are "divided" into three groups, each associated with specific types of attributes. 

* The TUD (TD Under Debug) group is associated with bits 0:7 and if any of them are set to 1 the TD is untrusted.
* The SEC (attributes that may impact security) group is associated with bits 8:31
* The Other (attributes not impacting security) group is the remaining bits 32:63